ui/component/or-rules/src/json-viewer/or-rule-action-attribute.ts

```diff
@@ -238,9 +249,48 @@ export class OrRuleActionAttribute extends translate(i18next)(LitElement) {
         this.requestUpdate();
     }
 
-    protected loadAssets(type: string) {
-        this.assetProvider(type).then(assets => {
-            this._assets = assets;
-        })
+    /**
+     * Fetches assets using the {@link assetProvider} from the parent component.
+     * This is often linked to the OpenRemote HTTP API to request assets from using an {@link AssetQuery} object.
+     * @param type - The asset type name to filter by
+     * @param search - The asset name to filter by (acts as a search)
+     * @param idValue - Selected asset ID to query along
+     * @protected
+     */
+    protected async loadAssets(type: string, search?: string, idValue?: string): Promise<Asset[] | undefined> {
```

Comment:
The `loadAssets` method could be instead a function reused between `or-rule-action-attribute` and `or-rule-asset-query`.

```suggestion
/**
 * Fetches assets using the {@link assetProvider} from the parent component.
 * This is often linked to the OpenRemote HTTP API to request assets from using an {@link AssetQuery} object.
 * @param type - The asset type name to filter by
 * @param search - The asset name to filter by (acts as a search)
 * @param idValue - Selected asset ID to query along
 * @protected
 */
function async loadAssets(type: string, provider: AssetProvider, search?: string, idValue?: string): Promise<Asset[] | undefined>
```

We can make these components implement the following interface and then pass `this` to the function.

```ts
export interface AssetProvider {
   provider: (type: string, query?: AssetQuery) => Promise<Asset[] | undefined>,
   cache: {query: AssetQuery, assets: Asset[]};
   loading: boolean
}
```

The only issue with this might be that the `cache` access scope might need to change which seems like a deal breaker.

Inheritance would then I guess be better.. let's discuss?